android/app/src/main/kotlin/com/jabook/app/jabook/audio/ExoPlayerMainThreadGuard.kt

```
@@ -0,0 +1,142 @@
+            }
+        }
+
+        val resultRef = AtomicReference<T>(null)
+        val errorRef = AtomicReference<Throwable>(null)
+        val latch = CountDownLatch(1)
+
+        val posted =
+            mainHandler.post {
+                try {
+                    resultRef.set(player.block())
+                } catch (e: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)